uiServices/src/testSuiteServices.go

```diff
@@ -30,58 +45,200 @@ func postTestSuites(rw http.ResponseWriter, req *http.Request) {
+		return
+
+	}
+
+	if !ValidateJsonWithSchema(buf.Bytes(),"testSuite_schema.json", "TestSuite") {
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	testSuite := testStrategies.TestSuite{}
+	err := json.Unmarshal(buf.Bytes(), &testSuite)
+	if err != nil {
+		logrus.Error("Cannot Unmarshall Json")
+		rw.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if !testSuiteWriterXml(testSuite, testSuitePathDir) {
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusNoContent)
+}
+
+func getTestSuite(rw http.ResponseWriter, req *http.Request){
+
+	testSuitePathDir := getTestSuiteHeader(req)
+	testSuiteName := chi.URLParam(req, "testSuiteName")
+
+	ValidateFileNameAndHeader(rw,req,testSuitePathDir, testSuiteName)
+
+	file, err := os.Open(fmt.Sprintf("%s%s.xml", testSuitePathDir, testSuiteName))
+	if err != nil {
+		logrus.Error("Test Suite Name Not Found: "+testSuitePathDir + testSuiteName)
+		rw.WriteHeader(http.StatusNotFound)
+		return
+	}
+	defer file.Close()
+
+
+	var testSuite testStrategies.TestSuite
+
+	byteValue, err := ioutil.ReadAll(file)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Read File")
+		return
+	}
+
+	err = xml.Unmarshal(byteValue, &testSuite)
+	if err != nil{
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Unmarshall")
+		return
+	}
+
+	testSuiteJSON, err := json.MarshalIndent(testSuite,"","")
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		logrus.Error("Cannot Marshall")
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+	rw.Write(testSuiteJSON)
+	logrus.Println(string(testSuiteJSON))
+
+}
+
+
+
+
+
+func getAllTestSuites(rw http.ResponseWriter, req *http.Request){
+	var testSuite testStrategies.TestSuite
+	var testSuiteList []testStrategies.TestSuite
+	var filename string
+	var filenameList []string
+
+	testSuitePathDir := getTestSuiteHeader(req)
+
+	files, err := ioutil.ReadDir(testSuitePathDir)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, file := range files {
+		if filepath.Ext(testSuitePathDir + file.Name()) == ".xml" {
+			filenameList = append(filenameList, file.Name())
+
+			filename = file.Name()
+
+			file, err := os.Open(fmt.Sprintf("%s%s", testSuitePathDir, filename))
+			if err != nil {
+				rw.WriteHeader(http.StatusInternalServerError)
+				logrus.Error("Cannot open file")
+				return
+			}
+
+			byteValue, err := ioutil.ReadAll(file)
+			if err != nil{
+				rw.WriteHeader(http.StatusInternalServerError)
+				logrus.Error("Cannot Read File")
+				return
+			}
+
+			err = xml.Unmarshal(byteValue, &testSuite)
+			if err != nil {
+				rw.WriteHeader(http.StatusInternalServerError)
+				logrus.Error("Cannot Unmarshall")
+				return
+			}
+
+			testSuiteList = append(testSuiteList, testSuite)
+		}
+	}
+
+	if len(filenameList) ==0{
+		logrus.Println("No Files Found")
+	}
+
+	json.NewEncoder(rw).Encode(filenameList)
+	json.NewEncoder(rw).Encode(testSuiteList)
```

Comment:
you are returning 2 separate lists here?

Why not one list which looks like:
```

[
    {
        "file": "testSuiteFile.xml",
        "name": "Team Test Suite",
        "description": "helpful description"
    }
]
```

With the current approach, it's a bit confusing because a client won't know which file matches which test suite.